test/unit_tests/braket/experimental/autoqasm/test_types.py

```diff
@@ -518,7 +518,7 @@ def retval_constant() -> int:
         return 3
 
     @aq.function
-    def retval_recursive() -> int:
+    def retval_recursive() -> float:
```

Comment:
Changed this because:
(a) `float` is actually the correct return type here (because of the `/ 3`).
(b) When this was `int`, the OpenQASM subroutine (after upgrading oqpy) was getting generated with a return type of `int` (machine-size) instead of `int[32]`. That's not necessarily a problem (it's still valid OQ3), but it alerted me that something weird was happening here. I still don't understand what changed in oqpy to cause this behavior change, but it works as expected if I give it the correct return type hint.

Reply:
I will open a task for myself to find a minimal repro.